File: scripts/reindex_news.py
import os
import re

news_dir = "src/data/news"
index_file = os.path.join(news_dir, "index.ts")

files = [f for f in os.listdir(news_dir) if f.endswith(".ts") and f != "index.ts"]
files.sort()

imports = []
articles = []

for i, filename in enumerate(files):
    var_name = f"a{i}"
    module_name = filename[:-3]
    imports.append(f"import {{ article as {var_name} }} from './{module_name}';")
    articles.append(var_name)

content = "\n".join(imports) + "\n\n"
content += "export const NEWS_LIST = [\n    " + ", ".join(articles) + "\n].sort((a, b) => b.id - a.id);\n\n"
content += "export const NEWS_ARTICLES = NEWS_LIST;\n"

with open(index_file, "w") as f:
    f.write(content)

print(f"Successfully reindexed {len(files)} articles.")

File: scripts/reindex-news.ts
import fs from 'fs';
import path from 'path';

const newsDir = 'src/data/news';
const files = fs.readdirSync(newsDir)
    .filter(f => f.endsWith('.ts') && f !== 'index.ts');

const imports: string[] = [];
const variables: string[] = [];

files.forEach((file, index) => {
    const name = file.replace('.ts', '');
    imports.push(`import { article as a${index} } from './${name}';`);
    variables.push(`a${index}`);
});

const content = `${imports.join('\n')}

export const NEWS_LIST = [
    ${variables.join(', ')}
].sort((a, b) => b.id - a.id);

export const NEWS_ARTICLES = NEWS_LIST;
`;

fs.writeFileSync(path.join(newsDir, 'index.ts'), content);
console.log('index.ts updated with ' + files.length + ' articles.');
